src/state/lottery/fetchUnclaimedUserRewards.ts

```diff
@@ -0,0 +1,209 @@
+
+const fetchCakeRewardsForTickets = async (
+  winningTickets: LotteryTicket[],
+): Promise<{ ticketsWithUnclaimedRewards: LotteryTicket[]; cakeTotal: BigNumber }> => {
+  const calls = winningTickets.map((winningTicket) => {
+    const { roundId, id, rewardBracket } = winningTicket
+    return {
+      name: 'viewRewardsForTicketId',
+      address: lotteryAddress,
+      params: [roundId, id, rewardBracket],
+    }
+  })
+  const cakeRewards = await multicallv2(lotteryV2Abi, calls)
+
+  const cakeTotal = cakeRewards.reduce((accum: BigNumber, cakeReward: ethers.BigNumber[]) => {
+    return accum.plus(new BigNumber(cakeReward[0].toString()))
+  }, BIG_ZERO)
+
+  const ticketsWithUnclaimedRewards = winningTickets.map((winningTicket, index) => {
+    return { ...winningTicket, cakeReward: cakeRewards[index] }
+  })
+  // TODO: Remove log. To help in testing when verifying winning tickets.
+  console.log('winning tickets: ', ticketsWithUnclaimedRewards)
+  return { ticketsWithUnclaimedRewards, cakeTotal }
+}
+
+const getRewardBracketByNumber = (ticketNumber: string, finalNumber: string): number => {
+  // Winning numbers are evaluated right-to-left in the smart contract, so we reverse their order for validation here:
+  // i.e. '1123456' should be evaluated as '6543211'
+  const ticketNumAsArray = ticketNumber.split('').reverse()
+  const winningNumsAsArray = finalNumber.split('').reverse()
+  const matchingNumbers = []
+
+  // The number at index 6 in all tickets is 1 and will always match, so finish at index 5
+  for (let index = 0; index < winningNumsAsArray.length - 1; index++) {
+    if (ticketNumAsArray[index] !== winningNumsAsArray[index]) {
+      break
+    }
+    matchingNumbers.push(ticketNumAsArray[index])
+  }
+
+  // Reward brackets refer to indexes, 0 = 1 match, 5 = 6 matches. Deduct 1 from matchingNumbers' length to get the reward bracket
+  const rewardBracket = matchingNumbers.length - 1
+  return rewardBracket
+}
+
+export const getWinningTickets = async (
+  roundDataAndUserTickets: RoundDataAndUserTickets,
+): Promise<LotteryTicketClaimData> => {
+  const { roundId, userTickets, finalNumber } = roundDataAndUserTickets
+
+  const ticketsWithRewardBrackets = userTickets.map((ticket) => {
+    return {
+      roundId,
+      id: ticket.id,
+      number: ticket.number,
+      status: ticket.status,
+      rewardBracket: getRewardBracketByNumber(ticket.number, finalNumber),
+    }
+  })
+  // TODO: Remove log. To help in testing when verifying winning tickets.
+  console.log(`all tickets round #${roundId}`, ticketsWithRewardBrackets)
+
+  // A rewardBracket of -1 means no matches. 0 and above means there has been a match
+  const allWinningTickets = ticketsWithRewardBrackets.filter((ticket) => {
+    return ticket.rewardBracket >= 0
+  })
+
+  // If ticket.status is true, the ticket has already been claimed
+  const unclaimedWinningTickets = allWinningTickets.filter((ticket) => {
+    return !ticket.status
```

Comment:
ignore for now: I guess `status` can be renamed to `claimed`? no need to refactor now if it takes long time I guess, but could be on todo list